Upgrade: 1.1.2 to 1.2.0 (#381)
harvester/docs: docs/upgrade/v1-1-2-to-v1-2-0.md

---
sidebar_position: 2
sidebar_label: Upgrade from v1.1.2 to v1.2.0
title: "Upgrade from v1.1.2 to v1.2.0"
---

<head>
  <link rel="canonical" href="https://docs.harvesterhci.io/v1.2/upgrade/v1-1-2-to-v1-2-0"/>
</head>

:::danger

> Please do not upgrade a running cluster to v1.2.0 if your machine has an **Intel E810** NIC card. We saw some reports that the NIC card has a problem when added to a bonding device. Please check this issue for more info: https://github.com/harvester/harvester/issues/3860.

Author: Removed the danger admonitions, e810 issue is fixed:
https://github.com/harvester/harvester/issues/3860#issuecomment-1687322977

:::

## General information

:::tip

Before you start an upgrade, you can run the pre-check script to make sure the cluster is in a stable state. For more details, please visit this [URL](https://github.com/harvester/upgrade-helpers/tree/main/pre-check/v1.1.x) for the script.
:::

Once there is an upgradable version, the Harvester GUI Dashboard page will show an upgrade button. For more details, please refer to [start an upgrade](./automatic.md#start-an-upgrade).

For the air-gap env upgrade, please refer to [prepare an air-gapped upgrade](./automatic.md#prepare-an-air-gapped-upgrade).

> ## Known Issues

Reviewer: ```suggestion
## Known issues
```

Author: Thank you! Fixed!

---

> ### 1. The upgrade can't start because of `"validator.harvesterhci.io" denied the request: managed chart rancher-monitoring is not ready, please wait for it to be ready`

Reviewer: ```suggestion
### 1. An upgrade can't start and reports `"validator.harvesterhci.io" denied the request: managed chart rancher-monitoring is not ready, please wait for it to be ready`
```

If a cluster is configured with a **storage network**, an upgrade can't start with the following message.

![](/img/v1.2/upgrade/known_issues/3839-error.png)

> - Related issue
>   - [[Doc] upgrade stuck while upgrading system service with alertmanager and prometheus](https://github.com/harvester/harvester/issues/3839)
> - Workaround

Reviewer: ```suggestion
- Related issue:
  - [[Doc] upgrade stuck while upgrading system service with alertmanager and prometheus](https://github.com/harvester/harvester/issues/3839)
- Workaround:
```

  - https://github.com/harvester/harvester/issues/3839#issuecomment-1534438192

---

> ### 2. An upgrade is stuck on creating the upgrade repository

Reviewer: ```suggestion
### 2. An upgrade is stuck in `Creating Upgrade Repository`
```

> An upgrade is pending while creating upgrade the repository:

Reviewer: ```suggestion
During an upgrade, **Creating Upgrade Repository** is stuck in the **Pending** state:
```

  ![](/img/v1.2/upgrade/known_issues/4246-pending.png)

> Please do the following operations to check if the cluster runs into the issue:

Reviewer: ```suggestion
Please perform the following steps to check if the cluster runs into the issue:
```

1. Check the upgrade repository pod:

    ![](/img/v1.2/upgrade/known_issues/4246-upgrade-repo-pod.png)

>     The `virt-launcher-upgrade-repo-hvst-<upgrade-name>` pod stays in `ContainerCreating`.

Reviewer: ```suggestion
    If the `virt-launcher-upgrade-repo-hvst-<upgrade-name>` pod stays in `ContainerCreating`, your cluster might have run into this issue. In this case, proceed with step 2.
```
